crates/worker/src/partition/mod.rs

```diff
@@ -470,6 +470,17 @@ where
         // ignore errors that happen during shut down
         let _ = self.leadership_state.step_down().await;
 
+        // Drain rpc_rx
+        self.rpc_rx.close();
+        while let Some(msg) = self.rpc_rx.recv().await {
+            let _ = msg
+                .into_outgoing(Err(PartitionProcessorRpcError::NotLeader(
+                    self.partition_id,
+                )))
+                .send()
+                .await;
+        }
```

Comment:
I think this is not enough since we can also leave this method via an `Error`. Given that it is not guaranteed how the error will be handled (panic vs. not), we probably should try to fail these requests also in this case.